PhysicsTools/PyTorch/interface/AlpakaConfig.h

```diff
@@ -0,0 +1,236 @@
+ * 
+ * @tparam T The type of Alpaka object (Device or Queue).
+ * @param obj The Alpaka object (Device or Queue) to convert.
+ * @return Corresponding PyTorch device.
+ */
+  template <typename T>
+  inline ::torch::Device device(const T &obj) {
+    if constexpr (::alpaka::isDevice<T>)
+      return ::torch::Device(kTorchDeviceType, obj.getNativeHandle());
+    else if constexpr (::alpaka::isQueue<T>)
+      return ::torch::Device(kTorchDeviceType, ::alpaka::getDev(obj).getNativeHandle());
+    else
+      static_assert(false_value<T>, "Unsupported type passed to device()");
+  }
+
+  /**
+   * @brief Computes a unique hash representation for the given Alpaka queue.
+   * 
+   * This function generates a hash string that uniquely represents the given Alpaka queue.
+   * For debugging purposes, to identify different queues in multi-threaded environments.
+   * 
+   * @tparam TQueue The type of Alpaka queue.
+   * @param queue The Alpaka queue to generate the hash for.
+   * @return A string representing the unique hash for the queue.
+   */
+  template <typename TQueue, typename = std::enable_if_t<::alpaka::isQueue<TQueue>>>
+  inline std::string queue_hash(const TQueue &queue) {
+    std::stringstream repr;
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    auto stream = c10::cuda::getStreamFromExternal(queue.getNativeHandle(), device(queue).index());
+    repr << "0x" << std::hex << stream.stream();
+    return repr.str();
+// #elif ALPAKA_ACC_GPU_HIP_ENABLED
+//   return "0x0";
+#endif
+    repr << "0x" << std::hex << std::hash<std::thread::id>{}(std::this_thread::get_id());
+    return repr.str();
+  }
+
+  /**
+   * @brief Computes a unique hash representation for the current stream associated with the given Alpaka queue.
+   * 
+   * This function generates a hash string representing the current stream of the given Alpaka queue,
+   * For debugging purposes - identifying the currently active stream.
+   * 
+   * @tparam TQueue The type of Alpaka queue.
+   * @param queue The Alpaka queue to generate the current stream hash for.
+   * @return A string representing the unique hash for the current stream.
+   */
+  template <typename TQueue, typename = std::enable_if_t<::alpaka::isQueue<TQueue>>>
+  inline std::string current_stream_hash(const TQueue &queue) {
+    std::stringstream repr;
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    const auto dev = device(queue);
+    auto stream = c10::cuda::getCurrentCUDAStream(dev.index());
+    repr << "0x" << std::hex << stream.stream();
+    return repr.str();
+// #elif ALPAKA_ACC_GPU_HIP_ENABLED
+//     return "0x0";
+#endif
+    repr << "0x" << std::hex << std::hash<std::thread::id>{}(std::this_thread::get_id());
+    return repr.str();
+  }
+
+  /**
+   * @brief Sets the guard to disable multi-threading and control PyTorch's threading model.
+   * @note Global call.
+   * 
+   * TODO: this should be called only once to disable PyTorch multi-threading, 
+   *       consider moving to AlpakaService.
```

Comment:
As mentioned today: cannot be `AlpakaService` because it would add a dependency on PyTorch, and because it should be usable also by PyTorch-based modules that do not use alpaka.
How about a new `PyTorchService` ?